### DISCUSSION TITLE
Invalidated Channel Openning Transaction Stays Pending

### DISCUSSION THREAD

**Discussion by AhmedMSedeek** [Ride-The-Lightning/RTL#843] [General]
Hello,

I used RTL to open several consecutive channels using unconfirmed UTXOs, after a while of waiting for all transactions to be confirmed, I decided to increase the fee using RBF feature, but instead of bumping the fee for the last unconfirmed transaction, I accidentally bumped the fee for the preceding unconfirmed transaction, resulting to invalidating the last channel creation unconfirmed UTXO

This creating the following two problem:

1. **Temporary**: The invalidated UTXO which is returning to the wallet does not disappear from the unconfirmed UTXOs list and On-chain balance
	This leads to incorrect higher On-chain balance which can again be spent and cause additional problems
	This issue disappears when the fee-bumping transaction is confirmed on the blockchain

2. **Permanent**: The channel that was supposed to be opened with the invalidated transaction stays in the "Pending Open" channel list

**To Reproduce**
1. Go to 'Lightning' => 'Peers/Channels'
2. Click on 'Open Channel'
3. Fill the required information and 
4. Click on "Open Channel", we will call this "Channel 1" for reference
5. Repeat steps from 1 to 3 with different information
6. Make sure to enter amount that is higher than your confirmed on-chain balance and lower than your total on-chain balance
7. Click on "Advanced Options"
8. Check "Spend Unconfirmed Output"
9. Click on "Open Channel", we will call this "Channel 2" for reference
10. Return to 'Lightning' => 'Peers/Channels' screen
11. Click on "Pending" tab
12. Expand "Pending Open" group
13. Next to "Channel 1", Open "Actions" drop down list
14. Select "Bump Fee" from the list
15. Fill the required information targeting to decrease the change address output
16. Click on "Bump Fee"

**Your environment**
* Version of `RTL` 0.11.2-beta
* Version of `lnd` 0.13.3-beta
* Version of `bitcoind` 22.0.0
* Browser Firefox, browser version 93
* Operating system Windows Server 2019

**On-Chain Tracking**
"Channel 1" Opening Transaction:
[https://www.blockchain.com/btc/tx/c5b5d85d5c63ba2fcc2997c7a123183dedf4d38d19af90f6ca9a3db5ef78b5e4](https://www.blockchain.com/btc/tx/c5b5d85d5c63ba2fcc2997c7a123183dedf4d38d19af90f6ca9a3db5ef78b5e4)

"Channel 2" Opening Transaction: **(Invalidated)**
[https://www.blockchain.com/btc/tx/e1684ff45f878b634282cef82650cac63a7ac9e5b3eaec73796d5773fb3b4db5](https://www.blockchain.com/btc/tx/e1684ff45f878b634282cef82650cac63a7ac9e5b3eaec73796d5773fb3b4db5)

Fee-Bumping Transaction:
[https://www.blockchain.com/btc/tx/ccdf54d2e4757b6145597f5a0eec8266429815fb7b17a34e6205d2099c789b29](https://www.blockchain.com/btc/tx/ccdf54d2e4757b6145597f5a0eec8266429815fb7b17a34e6205d2099c789b29)

**Suggested Solutions**
I can think of two ways of solving this:

1. Reflect the invalidated transaction on the application so that the invalid balance disappears from unconfirmed balance and the channel is removed from the "Pending Open" channels list.
2. An even better solution is to prevent the "Bump Fee" option for unconfirmed channel opening transaction with output(s) that has already been spent by another transaction, that means I can bump the fees for leaf transactions only.

**Comment by saubyk** [comment]
Moving this to discussions, will comment once I have assessed your comments.

**Comment by AhmedMSedeek** [comment]
I digged a little deeper and found out this is probably lnd issue, because lnd itself is currently reporting the invalidated channel opening request as pending
